### PR TITLE
TFileManager, TFileSystem, complete TTRB & more

### PR DESCRIPTION
**Pull request by InfiniteC0re**
**Added:**

Misc:

- TWIP macros
- Data folder (OpenBlob/Data)
- Started work on ALocaleManager
- SYMB and SecInfo [] operators

Methods:

- TFileManager::MountFileSystem
- TFileManager::SetSystemPath
- TFileManager::FindFileSystem
- TFileManager::CreateFile
- TFileManager::Initialize
- TFileManager::Destroy
- TFileManager::ValidateSystemPath
- TFileSystem::UnmountFileSystem
- TFile::Destroy
- TFile::Create
- TGenericDList::Head
- TGenericDList::Tail
- TGenericDList::Begin
- TGenericDList::End
- TGenericDList::IsEmpty
- TGenericDList::IsLinked
- TGenericDList::RemoveHead
- TGenericDList::RemoveTail

### CODE REVIEW

File: Toshi/Source/Toshi/Strings/TCString.cpp [hunk 1]
```diff
@@ -103,26 +103,24 @@ namespace Toshi
...
-	uint32_t TCString::Find(char character, uint32_t pos) const
+	int32_t TCString::Find(char character, uint32_t pos) const
 	{
 		if (!IsIndexValid(pos)) return -1;
 
-		TASSERT(IsIndexValid(0), "IsIndexValid(0)");
-
 		const char* foundAt = strchr(&m_pBuffer[pos], character);
-		if (foundAt == TNULL) return -1;
+		if (foundAt == nullptr) return -1;
```
Comment (AdventureT) [inline Toshi/Source/Toshi/Strings/TCString.cpp:111]: The typedef of tnull is nullptr

Comment (InfiniteC0re) [reply Toshi/Source/Toshi/Strings/TCString.cpp:111]: yep, it is, but strchr isn't related to Toshi in any way and that's why I think it's better to use TNULL when dealing with custom methods and default nullptr and win32's NULL value when dealing with standard methods like this one. you can also notice that in some of the TASSERTs they used NULL to compare with values from fe DirectInput (008c61a8) or T2Mutex_Win (0x008e8bc4) or TThread (008c5530)

Comment (AdventureT) [reply Toshi/Source/Toshi/Strings/TCString.cpp:111]: Hm, makes sense

File: Toshi/Source/Toshi/File/TTRB.cpp [hunk 2]
```diff
@@ -64,45 +56,40 @@ namespace Toshi
...
 					m_pHeader->m_ui32Version = 0;
 
 					ttsf.ReadBytes(&m_pHeader->m_i32SectionCount, sizeof(m_pHeader->m_i32SectionCount));
 
-					// TASSERT(m_pHeader->m_ui32Version == TMAKEVERSION(0, 0), "HEAD section cannot have a version");
 					TASSERT(m_pHeader->m_i32SectionCount == numsections, "HEAD section has wrong num of sections");
 
-					SecInfo* pCur = reinterpret_cast<SecInfo*>(m_pHeader + 1);
-					for (size_t i = 0; i < m_pHeader->m_i32SectionCount; i++)
+					SecInfo* pSect = &m_pHeader->operator[](0);
```
Comment (AdventureT) [inline Toshi/Source/Toshi/File/TTRB.cpp]: &m_pHeader[0]?

Comment (InfiniteC0re) [reply Toshi/Source/Toshi/File/TTRB.cpp]: operator[] doesn't work with pointers

Comment (AdventureT) [reply Toshi/Source/Toshi/File/TTRB.cpp]: Hm then i would call it GetSection(int index) or smth like that. In my opinion that would look better

File: Toshi/Source/Toshi/File/TTRB.cpp [hunk 3]
```diff
@@ -108,7 +108,7 @@ namespace Toshi
 				{
 					for (int i = 0; i < m_pHeader->m_i32SectionCount; i++)
 					{
-						auto& secInfo = m_pHeader->operator[](i);
+						auto& secInfo = (*m_pHeader)[i];
```
Comment (AdventureT) [inline Toshi/Source/Toshi/File/TTRB.cpp]: That's fine but what do you think about GetSectionInfo(int index) instead of this

Comment (InfiniteC0re) [reply Toshi/Source/Toshi/File/TTRB.cpp]: done

**Pull request merged**